server/master/src/main/java/org/apache/accumulo/master/FateServiceHandler.java

```diff
@@ -664,4 +670,41 @@ private String validateNamespaceArgument(ByteBuffer namespaceArg, TableOperation
           TableOperationExceptionType.INVALID_NAME, why);
     }
   }
+
+  /**
+   * Create a file on the file system to hold the splits to be created at table creation.
+   */
+  private String createSplitFile(final long opid, final List<ByteBuffer> arguments,
```

Comment:
This is interesting and seems like a first for fate transactions.  I am just curious of your thoughts for storing the splits in a file in tmp?  Wouldn't this be a problem with recovery if the server is restarted in the middle of the fate transaction and the file is gone?

Reply:
I went with the writing of splits to a temp file on the filesystem  after some discussions with @keith-turner. Given the size of some of the split sets it wouldn't be feasible to pass around the data within the FATE operations. Keith suggested writing them to a temporary file then then reading them in during the FATE operation itself. You are correct that there could be an issue if the server went down before the file was written, I think we decided that the odds of that happening would be low (hopefully).  I don't have a better solution at the moment. I'm open to alternative suggestions.

Reply:
@milleruntime were you thinking this is `/tmp` on the local FS?  I think this may `/tmp` in an accumulo hdfs dir.

Reply:
> You are correct that there could be an issue if the server went down before the file was written

Not sure there is a problem with the server going down.   The file is created before the FATE op is seeded.  If the servers goes down while its trying to setup the FATE op, then the client will retry RPC to seed fate op and in the process resend the splits.  Need to ensure the code that creates the split file is idempotent.  If the code is not idempotent, that could cause problems.